mod/wikirate_source/set/type_plus_right/source/file/import.rb

```diff
@@ -1,237 +1,62 @@
+include Card::Set::Abstract::Import
 
-event :validate_import, :prepare_to_validate,
-      on: :update,
-      when: proc { Env.params['is_metric_import_update'] == 'true' } do
-  metric_pointer_card = subcard(cardname.left + "+#{Card[:metric].name}")
-  metric_year = subcard(cardname.left + "+#{Card[:year].name}")
+def metric_pointer_card
+  subcards[cardname.left + "+#{Card[:metric].name}"]
+end
 
-  if !metric_pointer_card ||
-     !(metric_card = metric_pointer_card.item_cards.first)
-    errors.add :content, 'Please give a metric.'
-  elsif metric_card.type_id != Card::MetricID
-    errors.add :content, 'Invalid metric'
-  end
+def metric_year_card
+  subcard(cardname.left + "+#{Card[:year].name}")
+end
 
-  if !metric_year || !(year_card = metric_year.item_cards.first)
-    errors.add :content, 'Please give a year.'
-  elsif year_card.type_id != Card::YearID
-    errors.add :content, 'Invalid Year'
-  end
+def metric_content
+  metric_pointer_card.item_names.first
+end
+
+def metric_year_content
+  metric_year_card.item_names.first
 end
 
-event :import_csv, :prepare_to_store,
+event :validate_import, :prepare_to_validate,
       on: :update,
       when: proc { Env.params['is_metric_import_update'] == 'true' } do
```

Comment:
Why do we have this 'is_metric_import_update' flag? Do we update import files without intending to import something?